internal/stylance-core/src/lib.rs

```diff
@@ -142,37 +145,46 @@ pub fn load_and_modify_css(
     css_file: &Path,
     config: &Config,
 ) -> anyhow::Result<ModifyCssResult> {
-    let hash_str = make_hash(manifest_dir, css_file, config.hash_len)?;
+    let hash = make_hash(manifest_dir, css_file, config.hash_len)?;
     let css_file_contents = fs::read_to_string(css_file)?;
 
-    let fragments = parse::parse_css(&css_file_contents).map_err(|e| anyhow!("{e}"))?;
+    let (contents, _) = create_new_css(&css_file_contents, &config.class_name_pattern, &hash)
+        .map_err(|e| anyhow!("{e}"))?;
+
+    Ok(ModifyCssResult {
+        path: css_file.to_owned(),
+        normalized_path_str: normalized_relative_path(manifest_dir, css_file)?,
+        hash,
+        contents,
+    })
+}
+
+pub fn create_new_css<'a>(
+    css: &'a str,
+    class_name_pattern: &ClassNamePattern,
+    hash: &str,
+) -> Result<(String, HashMap<&'a str, Cow<'a, str>>), ParseError<&'a str, ContextError>> {
+    let fragments = parse::parse_css(&css)?;
 
-    let mut new_file = String::with_capacity(css_file_contents.len() * 2);
-    let mut cursor = css_file_contents.as_str();
+    let mut class_names = HashMap::new();
```

Comment:
Is the data stored in a hashmap useful to your goals? If not I'd prefer this was a Vec.